cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/store/SourceControlMetadataStoreTest.java

```diff
@@ -0,0 +1,109 @@
+import io.cdap.cdap.spi.data.transaction.TransactionRunners;
+import java.time.Instant;
+import org.junit.Before;
+import org.junit.Test;
+
+public abstract class SourceControlMetadataStoreTest {
+
+  private static final String NAMESPACE = "testNamespace";
+  private static final String COMMIT_ID = "testCommitId";
+  private static final String SPEC_HASH = "testSpecHash";
+  private static final Instant LAST_MODIFIED = Instant.now();
+  private static final String NAME = "testName";
+  private static final int INITIAL_NUMBER_OF_ROWS = 1;
+  private static final int UPDATED_NUMBER_OF_ROWS = 2;
+  private static final ApplicationId APP_ID = new ApplicationId(NAMESPACE, NAME);
+  private static final SourceControlMeta SOURCE_CONTROL_META = new SourceControlMeta(SPEC_HASH,
+      COMMIT_ID, LAST_MODIFIED);
+
+  protected static TransactionRunner transactionRunner;
+
+  @Before
+  public void before() {
+    TransactionRunners.run(transactionRunner, context -> {
+      SourceControlMetadataStore store = SourceControlMetadataStore.create(context);
+      store.deleteNamespaceSourceControlMetadataTable();
+    });
+
+    TransactionRunners.run(transactionRunner, context -> {
+      SourceControlMetadataStore store = SourceControlMetadataStore.create(context);
+      store.setNamespaceSourceControlMeta(APP_ID, SOURCE_CONTROL_META);
+    });
+  }
+
+  @Test
+  public void testSetNamespaceSourceControlMeta() throws Exception {
+    TransactionRunners.run(transactionRunner, context -> {
+      SourceControlMetadataStore store = SourceControlMetadataStore.create(context);
+      store.setNamespaceSourceControlMeta(new ApplicationId(NAMESPACE, "test2"), null);
+      long count = store.countNamespaceSourceControlMetadataRecords();
+      assertEquals(UPDATED_NUMBER_OF_ROWS, count);
+    });
+  }
+
+  @Test
+  public void testGetNamespaceSourceControlMeta() throws Exception {
+    TransactionRunners.run(transactionRunner, context -> {
+      SourceControlMetadataStore store = SourceControlMetadataStore.create(context);
+      SourceControlMeta scmMeta = store.getNamespaceSourceControlMeta(APP_ID);
+      long count = store.countNamespaceSourceControlMetadataRecords();
+      assertEquals(INITIAL_NUMBER_OF_ROWS, count);
+      assertEquals(SOURCE_CONTROL_META, scmMeta);
+    });
+  }
+
+  @Test
+  public void testUpdateNamespaceSourceControlMeta() throws Exception {
+    SourceControlMeta newSourceControlMeta = new SourceControlMeta("newFileHash", "newCommitId",
+        Instant.now());
+    TransactionRunners.run(transactionRunner, context -> {
+      SourceControlMetadataStore store = SourceControlMetadataStore.create(context);
+      store.updateNamespaceSourceControlMeta(APP_ID, newSourceControlMeta);
+      long count = store.countNamespaceSourceControlMetadataRecords();
+      SourceControlMeta scmMeta = store.getNamespaceSourceControlMeta(APP_ID);
+
+      assertEquals(newSourceControlMeta, scmMeta);
+      assertEquals(INITIAL_NUMBER_OF_ROWS, count);
```

Comment:
no need for this check. Get should always return one row.

Reply:
Got it. Done